src/App.tsx

```diff
@@ -1,156 +1,80 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Filter } from './components/Filter';
+import { TodoList } from './components/TodoList';
+import { Notification } from './components/Notification';
+import { AddTodo } from './components/AddTodo';
```

Comment:
to avoid a lot of imports, you can add `index.ts` file in your `component` folder, export each component, and import like
```
import { Filter, TodoList, Notification, AddTodo } from './components';
```